document-viewer/src/main/java/org/ebookdroid/common/provider/DocumentContentProvider.java

```diff
@@ -0,0 +1,54 @@
+package org.ebookdroid.common.provider;
+
+import android.content.ContentProvider;
+import android.content.ContentValues;
+import android.database.Cursor;
+import android.net.Uri;
+import android.os.ParcelFileDescriptor;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+
+public class DocumentContentProvider extends ContentProvider {
+
+    @Override
+    public ParcelFileDescriptor openFile(Uri uri, String mode) throws FileNotFoundException {
```

Comment:
I'm worried this content provider is a security vulnerability, it looks like any app could use it to read any file on the device? Going to look into it a bit more